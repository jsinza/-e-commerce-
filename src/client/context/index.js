import React, { createContext, useReducer } from 'react';
import { reducer, INITIAL_STATE } from '../reducers';
import { fetchFailure, fetchItemDetailStart, fetchItemDetailSuccess, fetchItemsStart, fetchItemsSuccess } from '../actions';
import * as itemService from '../services';

const StoreContext = createContext();

const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, INITIAL_STATE);
  const value = { state, dispatch };

  return (
    <StoreContext.Provider value={value}>
      {children}
    </StoreContext.Provider>
  );
};

const useStore = () => {
  const context = React.useContext(StoreContext);
  if (context === undefined) {
    throw new Error('useCount must be used within a CountProvider');
  }
  return context;
};

const searchItems = async (dispatch, query) => {
  dispatch(fetchItemsStart(query));
  try {
    const { data } = await itemService.searchItems(query);
    dispatch(fetchItemsSuccess(data));
  } catch ({ message }) {
    dispatch(fetchFailure(message));
  }
};

const getItem = async (dispatch, id) => {
  dispatch(fetchItemDetailStart());
  try {
    const { data } = await itemService.getItem(id);
    dispatch(fetchItemDetailSuccess(data));
  } catch ({ message }) {
    dispatch(fetchFailure(message));
  }
};

export { useStore, StoreProvider, searchItems, getItem };
