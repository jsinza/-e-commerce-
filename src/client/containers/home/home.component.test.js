
import React from 'react';
import { mount } from 'enzyme';
import 'babel-polyfill';
import Home from './home.component';
import * as context from '../../context';

jest.mock('../../context');

const LOCATION = {
  search: {
    q: 'zapatos',
  },

};

describe('Home Components', () => {
  let wrapper;

  beforeEach(() => {
    context.useStore.mockReturnValue({ dispatch: jest.fn(), state: { categories: [], items: [] } });
    wrapper = mount(<Home location={LOCATION} />);
  });

  it('should render Home component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the home container', () => {
    expect(wrapper.find("[className='home']")).toHaveLength(1);
  });

  it('should get the search parameters', () => {
    expect(context.searchItems.mock.calls[2][1]).toBe(LOCATION.q);
  });
});

