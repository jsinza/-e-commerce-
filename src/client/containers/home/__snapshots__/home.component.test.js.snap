// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Components should render Home component 1`] = `ReactWrapper {}`;
